### ISSUE TITLE
General Population does not save without another Demographic Indicator

### ISSUE THREAD

**Issue by lache-melvin** [msupply-foundation/open-msupply#4222]
## What went wrong? 😲

If you have no demographic indicators (only the `General Population` row) and try to update the population:

- hit save - looks like success
- refresh - it aint there

https://github.com/msupply-foundation/open-msupply/assets/55115239/c27085a9-7603-4a60-b75f-7a3e75fb5a96

This is because we store the general population against the other projection rows, rather than on its own. Will require a bit of rework to save this separately. I'm not sure where exactly?

## Expected behaviour 🤔

Gen population row saves, regardless of what other rows you have

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. See above 🙈 

## Your environment 🌱

- Open mSupply Version: 2.1 RC1
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by mark-prins** [comment]
I did wonder if we should save gen pop as a row in database, same as the others but with a specific ID so that we can customise behaviour. either insert as a migration or just upsert from f/e?

**Comment by lache-melvin** [comment]
Is there a future where we add ability to delete rows? Also wouldn't want that to be possible... but yeah with specific ID we could work around that 😁 

Almost wondered if it was a value you could store on the demographic_projection row? Maybe it's mixing concerns?

**Comment by mark-prins** [comment]
yeah - could add to `demographic_projection` but it is a little bit of a mixed concern. At least with the `demographic_indicator` row most of the data is applicable and it is shown in the table. It also means that we don't have to recalculate on first load - just show what's in the db

**Comment by Chris-Petty** [comment]
Low - no budget for GAPS area, no users to affect

**Comment by fergie-nz** [comment]
Can we add a separate general population table here?
A bit more upfront work with migrating / refactoring, but feels to me like similar issues will crop up with this current structure.

I was also wondering about adding an 'is_index' column to demographic_indicator table, and store general population in the same table. No migrations, but slightly messy in the constraints for implementation: 
- Only one index
- Can't delete index
Also
- Automatically creating an index per year?
- Maybe some other issues in differentiating logic between general pop and other population rows?

**Comment by lache-melvin** [comment]
Was looking at this a while ago, @jmbrunskill and I wondered about just having a known ID for this row? Similar idea to the is_index.

Kind of like we have for [system user id](https://github.com/msupply-foundation/open-msupply/blob/develop/server/util/src/constants.rs#L24)... I guess with the year tacked on the end?

Frontend can save the general population to this id/expect it to be there?

@andreievg maybe you have thoughts? Not sure if there's any word of wanting to do more things with general population values, maybe this would push us towards storing in some other way?

**Comment by lache-melvin** [comment]
@msupply-foundation/triage probably worth keeping this around/addressing before GAPS pilot?

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.18.00-develop-04100648`

- [x] Go to manage -> demographics
- [x] Don't add any new demographics, just edit the general one
- [x] Save
- [x] Go to another page and back
- [x] General demographic numbers should still be there..

**Comment by marthakeezy** [comment]
Pass :) Moving to Done! ✅
